src/lib/controllers/aic.rs

```diff
@@ -67,19 +73,51 @@ pub async fn update(
             }
         }
         Err(e) => match e {
-            sqlx::Error::RowNotFound => return HttpResponse::NotFound().finish(),
-            _ => return HttpResponse::InternalServerError().finish(),
+            sqlx::Error::RowNotFound => {
+                error_and_incr!(
+                    statsd,
+                    LogKey::AicRecordUpdateFailedNotFound,
```

Comment:
Not strictly necessary but my be useful to pass the error in here anyway so it's clear to a future debugger what exactly is going on.